components/gallery/GalleryItem.vue

```diff
@@ -96,6 +97,12 @@
                       </nuxt-link>
                     </template>
                   </CollectionDetailsPopover>
+                  <nuxt-link
+                    :to="`/${urlPrefix}/collection/${collection?.id}`"
+                    class="has-text-link block md:hidden"
+                    data-testid="gallery-item-collection-link">
+                    {{ collection?.name || collection?.id }}
+                  </nuxt-link>
```

Comment:
Ideally, this logic should be placed inside the `CollectionDetailsPopover` component so it can be reused everywhere

Reply:
@Jarsen136 done